symfony/plugins/orangehrmLeavePlugin/Api/HolidayAPI.php

```diff
@@ -0,0 +1,230 @@
+use OrangeHRM\Core\Api\V2\Endpoint;
+use OrangeHRM\Core\Api\V2\EndpointCollectionResult;
+use OrangeHRM\Core\Api\V2\EndpointResourceResult;
+use OrangeHRM\Core\Api\V2\EndpointResult;
+use OrangeHRM\Core\Api\V2\ParameterBag;
+use OrangeHRM\Core\Api\V2\RequestParams;
+use OrangeHRM\Core\Api\V2\Validator\ParamRule;
+use OrangeHRM\Core\Api\V2\Validator\ParamRuleCollection;
+use OrangeHRM\Core\Api\V2\Validator\Rule;
+use OrangeHRM\Core\Api\V2\Validator\Rules;
+use OrangeHRM\Entity\Holiday;
+use OrangeHRM\Leave\Api\Model\HolidayModel;
+use OrangeHRM\Leave\Dto\HolidaySearchFilterParams;
+use OrangeHRM\Leave\Service\HolidayService;
+
+class HolidayAPI extends Endpoint implements CrudEndpoint
+{
+    public const PARAMETER_NAME = 'name';
+    public const PARAMETER_DATE = 'date';
+    public const PARAMETER_RECURRING = 'recurring';
+    public const PARAMETER_LENGTH = 'length';
+
+    public const FILTER_FROM_DATE = 'fromDate';
+    public const FILTER_TO_DATE = 'toDate';
+
+    public const PARAM_RULE_NAME_MAX_LENGTH = 200;
+
+    private ?HolidayService $holidayService = null;
+
+    /**
+     * @return HolidayService
+     */
+    protected function getHolidayService(): HolidayService
+    {
+        if (!$this->holidayService instanceof HolidayService) {
+            $this->holidayService = new HolidayService();
+        }
+        return $this->holidayService;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getOne(): EndpointResult
+    {
+        $holiday = $this->getHolidayService()->getHolidayDao()->getHolidayById($this->getIdFromUrlAttributes());
+        $this->throwRecordNotFoundExceptionIfNotExist($holiday, Holiday::class);
+
+        return new EndpointResourceResult(HolidayModel::class, $holiday);
+    }
+
+    /**
+     * @return int
+     */
+    private function getIdFromUrlAttributes(): int
+    {
+        return $this->getRequestParams()->getInt(RequestParams::PARAM_TYPE_ATTRIBUTE, CommonParams::PARAMETER_ID);
+    }
+
+    /**
+     * @return ParamRule
+     */
+    private function getIdParamRule(): ParamRule
+    {
+        return new ParamRule(CommonParams::PARAMETER_ID, new Rule(Rules::POSITIVE));
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRuleForGetOne(): ParamRuleCollection
+    {
+        return new ParamRuleCollection($this->getIdParamRule());
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getAll(): EndpointResult
+    {
+        $fromDate = $this->getRequestParams()->getDateTime(RequestParams::PARAM_TYPE_QUERY, self::FILTER_FROM_DATE);
+        $toDate = $this->getRequestParams()->getDateTime(RequestParams::PARAM_TYPE_QUERY, self::FILTER_TO_DATE);
+
+        $holidaySearchFilterParams = new HolidaySearchFilterParams();
+        $holidaySearchFilterParams->setFromDate($fromDate);
+        $holidaySearchFilterParams->setToDate($toDate);
+        $holidays = $this->getHolidayService()->searchHolidays($holidaySearchFilterParams);
```

Comment:
Can't we chain setters 
`$holidaySearchFilterParams = (new HolidaySearchFilterParams())->setFromDate()->setToDate()`